pydm/widgets/analog_indicator.py

```diff
@@ -0,0 +1,1576 @@
+           self._upper_limit - self._lower_limit == 0:
+            proportion = -1 # Invalid
+        else:
+            proportion = (value - self._lower_limit) / (self._upper_limit - self._lower_limit)
+
+        position = int(proportion * self._widget_width)
+        return position
+
+    def set_origin(self):
+        """
+        Set the position (pixel) in which the origin should be drawn.
+        """
+        if self._origin_at_zero:
+            self._origin_position = self.calculate_position_for_value(0)
+        else:
+            self._origin_position = 0
+
+    def set_position(self):
+        """
+        Set the position (pixel) in which the pointer should be drawn.
+        """
+        self.position = self.calculate_position_for_value(self._value)
+
+    def update_indicator(self):
+        """
+        Update the position and the drawing of indicator.
+        """
+        self.set_position()
+        self.repaint()
+
+    def set_value(self, value):
+        """
+        Set a new current value for the indicator.
+        """
+        self._value = value
+        self.update_indicator()
+
+    def set_upper_limit(self, new_limit):
+        """
+        Set the scale upper limit.
+
+        Parameters
+        ----------
+        new_limit : float
+            The upper limit of the scale.
+        """
+        self._upper_limit = new_limit
+
+    def set_lower_limit(self, new_limit):
+        """
+        Set the scale lower limit.
+
+        Parameters
+        ----------
+        new_limit : float
+            The lower limit of the scale.
+        """
+        self._lower_limit = new_limit
+
+    def set_upper_minor_alarm(self, new_minor_alarm):
+        """
+        Set the scale upper minor alarm.
+
+        Parameters
+        ----------
+        new_minor_alarm : float
+            The upper minor alarm of the scale.
+        """
+        self._upper_minor_alarm = new_minor_alarm
+
+    def set_lower_minor_alarm(self, new_minor_alarm):
+        """
+        Set the scale lower minor alarm.
+
+        Parameters
+        ----------
+        new_minor_alarm : float
+            The lower minor alarm of the scale.
+        """
+        self._lower_minor_alarm = new_minor_alarm
+
+    def set_upper_major_alarm(self, new_major_alarm):
+        """
+        Set the scale upper major alarm.
+
+        Parameters
+        ----------
+        new_major_alarm : float
+            The upper major alarm of the scale.
+        """
+        self._upper_major_alarm = new_major_alarm
+
+    def set_lower_major_alarm(self, new_major_alarm):
+        """
+        Set the scale lower major alarm.
+
+        Parameters
+        ----------
+        new_major_alarm : float
+            The lower major alarm of the scale.
+        """
+        self._lower_major_alarm = new_major_alarm
+
+    def get_show_ticks(self):
+        return self._show_ticks
+
+    def set_show_ticks(self, checked):
```

Comment:
Seems like you stopped writing docstrings for this class at about here in the file- I don't blame you though, you've got a lot of method here!